lib/rouge/lexers/mmt.rb

```diff
@@ -74,10 +74,17 @@ class MMT < RegexLexer
       state :theoryHeader do
         rule %r/\s+/im, Text::Whitespace
 
-        rule %r/([^\s\u2759\u275a:=]+)(\s*)(?:(:)(\s*)([^\s\u2759\u275a=]+))?(\s*)(?:(>)([^\u2759\u275a=]+))?/im do
-          groups Name::Class,Text::Whitespace,Punctuation,Text::Whitespace,Text,Text::Whitespace,Punctuation,Name::Variable
-          goto :moduleDefiniens
+        rule %r/[^\s:=>]+/im, Name::Class
+
+        rule %r/(:)(\s*)([^\s=]+)/im do
+          groups Punctuation, Text::Whitespace, Text
         end
+
+        rule %r/(>)([^=]+)/im do
+          groups Punctuation, Name::Variable
+        end
+
+        rule(//) { goto :moduleDefiniens }
```

Comment:
@pyrmont I don't see how these changes are semantics-preserving. If `theoryHeader` is entered and the text at hand matches none of the rules except the last, we will go to `moduleDefiniens`. Overall, the specific structure *required* by the MMT language for theory headers is not enforced anymore.

Reply:
I should have been clearer about these changes. The fact that the specific structure required by MMT is not enforced is what I mean when I say that syntax correctness is a non-goal. We generally don't recommend that lexers include this logic if doing so complicates the lexer's rules as it creates space for bugs in the long regular expressions that tend to be required to do this.

If the changes cause syntactically _correct_ code to be lexed incorrectly then this is a mistake on my part. I didn't see any differences in the visual sample you included in the original PR. Were you concerned that that would be the result here?

Reply:
> We generally don't recommend that lexers include this logic if doing so complicates the lexer's rules

For me, using a state for a single purpose is easier than using a state for multiple purposes ("substates") as is the case after this change.
But I see where your point regarding longish regular expressions is coming from.

> If the changes cause syntactically correct code to be lexed incorrectly

I am not sure if this is the case, probably not, but I would need to test this on my large test base. (will do the next days)